File: php-pattern/basic/adapter/CarInfoAdapter.php
<?php

namespace app\adapter;

use app\adapter\adaptee\CarInfo;
use app\adapter\engine\Printable;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of CarInfoAdapter
 *
 * @author eko
 */
class CarInfoAdapter implements Printable {

    private $carInfo;

    function __construct(CarInfo $carInfo) {
        $this->carInfo = $carInfo;
    }

    public function printing() {
        echo $this->carInfo->toString();
    }

}

File: java-pattern/src/main/java/adapter/CarInfoAdapter.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package adapter;

import adapter.adaptee.CarInfo;
import adapter.engine.Printable;

/**
 *
 * @author eko
 */
public class CarInfoAdapter implements Printable {

    private final CarInfo carInfo;

    public CarInfoAdapter(CarInfo carInfo) {
        this.carInfo = carInfo;
    }

    @Override
    public String printing() {
        if (null != carInfo) {
            return carInfo.toString();
        }

        return "No object to be printed";
    }

}
